storage/postgres/postgres.go

```diff
@@ -97,6 +85,22 @@ func NewPostgresDatastore(uri string, opts ...PostgresOption) (*Postgres, error)
 		p.maxTypesInTypeDefinition = defaultMaxTypesInDefinition
 	}
 
+	var pool *pgxpool.Pool
+	err := backoff.Retry(func() error {
+		var err error
+		pool, err = pgxpool.Connect(context.Background(), uri)
+		if err != nil {
+			p.logger.Info("waiting for Postgres")
+			return err
+		}
+		return nil
+	}, backoff.NewExponentialBackOff())
```

Comment:
The default here uses an extremely long MaxElapsedTime setting (like 13min or something like that). We should create an exponential backoff policy with a tighter max elapsed time so that our readiness/liveness checks up front fail quicker if we can't connect to the database in time.

Reply:
Good point. Thanks! How does five minutes sound?

Reply:
I was thinking more like 30s - 1min max. Readiness checks in Cloud environments like Kubernetes are usually on the order of no more than a 2min timeout window. Setting a little tighter timeout here will make readiness a little more lively, which will help more quickly identify deployments that are wired up to a database that is unreachable.

Reply:
Okay. Let's try 1 minute ;)